worldedit-core/src/main/java/com/sk89q/worldedit/entity/Player.java

```diff
@@ -210,6 +211,17 @@
      */
     Location getBlockTrace(int range, boolean useLastBlock);
 
+    /**
+     * Get the point of the block being looked at. May return null.
+     * Will return the farthest away block before matching the stop mask if useLastBlock is true and no other block is found.
+     *
+     * @param range how far to checks for blocks
+     * @param useLastBlock try to return the last valid block not matching the stop mask found
+     * @param stopMask the mask used to determine when to stop tracing
+     * @return point
+     */
+    Location getBlockTrace(int range, boolean useLastBlock, Mask stopMask);
```

Comment:
Would it maybe make sense to make the `stopMask` nullable? Seems like there's a lot of branching around this call, when we could just pass it through.

Reply:
tbh i'm used to just making overrides for everything to maintain API compat.